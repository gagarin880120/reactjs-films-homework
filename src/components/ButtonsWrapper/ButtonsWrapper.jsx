import React, { useState } from 'react';
import PropTypes from 'prop-types';
import InfoButton from '../InfoButton/InfoButton';
import TrailerButtonRect from '../TrailerButtonRect/TrailerButtonRect';
import styles from './ButtonsWrapper.module.scss';

export default function ButtonsWrapper({ movie, onTrailerButtonClick }) {
  const [isInfoViewed, setIsInfoViewed] = useState(false);
  return (
    <div className={styles.wrapper}>
      <div
        className={styles.overview}
        style={{ display: isInfoViewed ? 'block' : 'none' }}
      >
        <p>{movie.overview}</p>
      </div>
      <div className={styles.buttonsContainer}>
        <TrailerButtonRect
          onTrailerButtonClick={onTrailerButtonClick}
          id={movie.id}
        />
        <InfoButton
          onInfoButtonClick={() => {
            if (isInfoViewed) {
              setIsInfoViewed(false);
            } else {
              setIsInfoViewed(true);
            }
          }}
        />
      </div>
    </div>
  );
}

ButtonsWrapper.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    overview: PropTypes.string,
  }),
  onTrailerButtonClick: PropTypes.func,
};

ButtonsWrapper.defaultProps = {
  movie: {
    id: 0,
    overview: '',
  },
  onTrailerButtonClick: null,
};
