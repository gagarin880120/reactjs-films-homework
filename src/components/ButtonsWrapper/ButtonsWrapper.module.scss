.wrapper {
  position: absolute;
  bottom: 40px;
  right: 2%;
  width: 250px;
  display: flex;
  justify-content: space-between;
}

.overview {
  padding: 1%;
  position: absolute;
  width: 140%;
  max-height: 280px;
  bottom: 55px;
  right: 0;
  background: rgba(0, 0, 0, 0.6);
  overflow-y: scroll;

  p {
    padding-left: 1%;
  }
}

.overview::-webkit-scrollbar {
  width: 8px;
}

.overview::-webkit-scrollbar-track {
  background: rgba(0, 0, 0, 0.5);
  border-radius: 6px;
}

.overview::-webkit-scrollbar-thumb {
  background: #47c3f0;
  border-radius: 6px;
}

.overview::-webkit-scrollbar-thumb:hover {
  background: #02729b;
}

@media screen and (max-width: 900px) {
  .overview {
    width: 100%;
    max-height: 240px;
  }
}

@media screen and (max-width: 750px) {
  .wrapper {
    bottom: 20px;
    left: 2%;
    flex-direction: column;
    height: 100px;
    width: 100%;

    button {
      width: 80px;
      height: 40px;
    }
  }

  .overview {
    max-height: 100px;
    width: 79%;
    overflow-y: scroll;
    left: 90px;
    bottom: 0;
  }
}

@media screen and (max-width: 400px) {
  .overview {
    width: 70%;
  }
}
