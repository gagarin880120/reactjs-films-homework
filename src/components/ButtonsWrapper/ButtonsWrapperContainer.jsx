import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ButtonsWrapper from './ButtonsWrapper';
import { movieDetailsSelector } from '../../redux/selectors';
import { setModal, getTrailer } from '../../redux/actions';

export function ButtonsWrapperContainer({ movie, onTrailerButtonClick }) {
  return (
    <ButtonsWrapper
      movie={movie}
      onTrailerButtonClick={onTrailerButtonClick}
    />
  );
}

export const mapStateToProps = (state) => ({
  movie: movieDetailsSelector(state),
});

export const mapDispatchToProps = (dispatch) => ({
  onTrailerButtonClick(id) {
    dispatch(setModal(true));
    dispatch(getTrailer(id));
  },
});

ButtonsWrapperContainer.propTypes = {
  movie: PropTypes.instanceOf(Object),
  onTrailerButtonClick: PropTypes.func,
};

ButtonsWrapperContainer.defaultProps = {
  movie: {},
  onTrailerButtonClick: null,
};

export default connect(mapStateToProps, mapDispatchToProps)(ButtonsWrapperContainer);
