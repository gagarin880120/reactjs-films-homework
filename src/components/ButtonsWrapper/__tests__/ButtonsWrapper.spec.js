import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import React from 'react';
import ButtonsWrapper from '../ButtonsWrapper';

describe('ButtonsWrapper component', () => {
  test('should render without crashing', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
    <ButtonsWrapper />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  describe('infoButton', () => {
    const testRenderer = TestRenderer.create(<ButtonsWrapper />);

    describe('when is clicked', () => {
      test('once, should render overview element', () => {
        const button = testRenderer.root.findByProps({className: 'infoButton'});

        act(() => {
          button.props.onClick();
        });

        const overview = testRenderer.root.findByProps({className: 'overview'});
        expect(overview.props.style.display).toBe('block');
      });

      test('two times, should not render overview element', () => {
        const button = testRenderer.root.findByProps({className: 'infoButton'});

        act(() => {
          button.props.onClick();
          button.props.onClick();
        });

        const overview = testRenderer.root.findByProps({className: 'overview'});
        expect(overview.props.style.display).toBe('none');
      });
    });
  });
});
