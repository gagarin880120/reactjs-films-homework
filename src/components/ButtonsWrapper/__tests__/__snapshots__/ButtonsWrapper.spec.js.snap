// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsWrapper component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <div
    className="overview"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      
    </p>
  </div>
  <TrailerButtonRect
    id={0}
    onTrailerButtonClick={null}
  />
  <InfoButton
    onInfoButtonClick={[Function]}
  />
</div>
`;
